app/services/renderer_methods.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# macros to fetch information about invoked jobs
+module RendererMethods
+  extend ActiveSupport::Concern
+
+  def find_job_invocation_by_id(job_id, preload: nil)
+    if (job =  JobInvocation.preload(preload).find_by(id: job_id).first)
+      job
+    else
+      raise ::Foreman::Exception.new(N_("Can't find Job Invocation for an id %s"), job_id)
+    end
+  end
+
+  def load_job_invocations(search, includes: nil, preload: nil)
```

Comment:
This is nice, but if that should be the only macro, I think using `load_resource` in the template would be good enough :man_shrugging:

Reply:
I've removed this macro. The main point for this macro was abstraction as suggested @ares in the begging of this PR. However I don't think that this should be use for daily basis so I removed them.